pkgs/development/tools/boost-build/default.nix

```diff
@@ -1,34 +1,39 @@
-{ lib, stdenv, fetchFromGitHub }:
+{ lib
+, stdenv
+, fetchFromGitHub
+, bison
+}:
 
 stdenv.mkDerivation rec {
   pname = "boost-build";
-  version = "2016.03";
+  version = "4.4.1";
 
   src = fetchFromGitHub {
     owner = "boostorg";
     repo = "build";
     rev = version;
-    sha256 = "1qw5marmp7z09nwcjlqrmqdg9b6myfqj3zvfz888x9mbidrmhn6p";
+    sha256 = "1r4rwlq87ydmsdqrik4ly5iai796qalvw7603mridg2nwcbbnf54";
   };
 
-  hardeningDisable = [ "format" ];
-
-  patchPhase = ''
-    grep -r '/usr/share/boost-build' \
-      | awk '{split($0,a,":"); print a[1];}' \
-      | xargs sed -i "s,/usr/share/boost-build,$out/share/boost-build,"
-  '';
+  nativeBuildInputs = [
+    bison
+  ];
 
   buildPhase = ''
+    runHook preBuild
     ./bootstrap.sh
+    runHook postBuild
   '';
 
   installPhase = ''
-    ./b2 install --prefix=$out
+    runHook preInstall
+    ./b2 install --prefix="$out"
+    ln -s b2 "$out/bin/bjam"
```

Comment:
```suggestion
    ln -s $out/bin/b2 $out/bin/bjam
```

Reply:
That really doesn't have any advantage.

Reply:
I don't think we need absolute path here. Symbolic link target is always related to itself. 
We also have `{cp,mv,ln} -> coreutils` in `coreutils`.

Reply:
Turns out the `fixupPhase` prefers relative symlinks anyway:
```
rewriting symlink /nix/store/z1cx1qsjqyx9dzxsxd67cn1jm3mwhc71-boost-build-4.4.1/bin/bjam to be relative to /nix/store/z1cx1qsjqyx9dzxsxd67cn1jm3mwhc71-boost-build-4.4.1
```